service/history/timerQueueActiveTaskExecutor.go

```diff
@@ -219,18 +219,23 @@ Loop:
 			break Loop
 		}
 
-		if timerSequenceID.timerType != timerTypeScheduleToStart {
```

Comment:
I removed this check because `RetryActivity` below will call `IsRetryable` and it will return `NonRetryableFailure` for `ScheduleToStart`.